### DISCUSSION TITLE
Multiple private-keys in a SOPS secret

### DISCUSSION THREAD

**Discussion by haraldatbmw** [fluxcd/flux2#456] [Q&A] [answered]
The guide for Mozilla SOPS says: 

> Note that the sops-gpg can contain more than one key, sops will try to decrypt the secrets by iterating over all the private keys until it finds one that works.

I think that is needed if I have multiple team-members that wants to encrypt secrets with SOPS using their local pgp-key.

It's not clear to my at which level I can put multiple private-keys in the sops-gpg secret.
Should I put multiple private-keys as the value of the key `sops.asc` or can I add multiple keys in the data section of the secret, each with one private-pgp-key?

**Comment by stefanprodan** [comment, marked as answer]
You can add multiple keys in the data section of the secret, each entry with its one private pgp key.

**Reply by haraldatbmw** [reply]
And the key-name is not relevant?

**Reply by stefanprodan** [reply]
Can be `<anything>.asc` but I would use the team member name

**Comment by kingdonb** [comment]
It still wasn't immediately clear to me how to add multiple keys to SOPS configuration after reading this thread, with multiple team members you want it configured so the message is encrypted to all keys (but only one key belongs to the flux agent, so that it can be rotated singularly.)

Found this link helpful, showing how one can easily add multiple keys to the same `creation_rules` entry in `.sops.yaml`, with a Block Scalar yaml multiline string: https://github.com/mozilla/sops#261updatekeys-command
